frontend2/src/api/loaders/episodeLoader.ts

```diff
@@ -0,0 +1,47 @@
+import { type QueryClient } from "@tanstack/react-query";
+import { ResponseError } from "api/_autogen";
+import { loginCheck } from "api/auth/authApi";
+import { episodeInfoFactory } from "api/episode/episodeFactories";
+import { buildKey, safeLoader } from "api/helpers";
+import { searchTeamsFactory, myTeamFactory } from "api/team/teamFactories";
+import { type LoaderFunction } from "react-router-dom";
+
+export const episodeLoader =
+  (queryClient: QueryClient): LoaderFunction =>
+  async ({ params }) => {
+    // check if the episodeId is a valid one.
+    // if the episode is not found, throw an error.
+    const id = params.episodeId ?? "";
+    if (id === "") {
+      throw new ResponseError(
+        new Response("Episode not found.", { status: 404 }),
+      );
+    }
+
+    // Run a check to see if the user has an invalid token
+    const loggedIn = await loginCheck(queryClient);
+
+    // Await the episode info so we can be sure that it exists
+    const episodeInfo = await queryClient.ensureQueryData({
```

Comment:
I'm still awaiting things that I want to e.g. wait or throw to error boundary on :sunglasses: